config/crd/bases/kci.rocks_databases.yaml

```diff
@@ -0,0 +1,298 @@
+                      monitoring:
+                        description: DbInstanceMonitoring defines if exporter
+                        properties:
+                          enabled:
+                            type: boolean
+                        required:
+                        - enabled
+                        type: object
+                      percona:
+                        description: PerconaCluster is used when instance type is
+                          percona cluster
+                        properties:
+                          monitorUserSecretRef:
+                            description: NamespacedName is required because CRD structs
+                              must have all fields json tagged.
+                            properties:
+                              Name:
+                                type: string
+                              Namespace:
+                                type: string
+                            required:
+                            - Name
+                            - Namespace
+                            type: object
+                          servers:
+                            items:
+                              description: BackendServer defines backend database
+                                server
+                              properties:
+                                host:
+                                  type: string
+                                maxConn:
+                                  type: integer
+                                port:
+                                  type: integer
+                                readonly:
+                                  type: boolean
+                              required:
+                              - host
+                              - maxConn
+                              - port
+                              type: object
+                            type: array
+                        required:
+                        - monitorUserSecretRef
+                        - servers
+                        type: object
+                      sslConnection:
+                        description: DbInstanceSSLConnection defines weather connection
+                          from db-operator to instance has to be ssl or not
+                        properties:
+                          enabled:
+                            type: boolean
+                          skip-verify:
+                            description: SkipVerity use SSL connection, but don't
+                              check against a CA
+                            type: boolean
+                        required:
+                        - enabled
+                        - skip-verify
+                        type: object
+                    required:
+                    - adminSecretRef
+                    - backup
+                    - engine
+                    - monitoring
+                    - sslConnection
```

Comment:
these are auto generated? one of my last PR was about removing some required fields by hand and setting`additionalProperties: true` some places. See: https://github.com/kloeckner-i/db-operator/pull/50/files#diff-ec73b463d58b152098494e2bd805cebe3c759b302f7585de5c0d6aaa2ab2470fR67
It's hard to see what was changed here. We have many places required properties are not set it might break pipeline. can we keep non breaking change? then how to mange it in the future?

Reply:
Yes this is autogenerated, if we want these fields to be optional we're going to need to take a look at how we configure controller-gen and any tags/comments we can use.

Thanks for the catch.

Reply:
but right now these files won't be used anywhere right? since we have help chart defining crd